apps/frontend/src/pages/project/project.tsx

```diff
@@ -83,6 +100,39 @@ const Project = (): JSX.Element => {
 		[dispatch, project],
 	);
 
+	const handleEditContributor = useCallback(
+		(contributorId: number) => {
+			const contributor = projectContributors.find(
+				(contributor) => contributor.id === contributorId,
+			);
+
+			if (contributor) {
+				setContributorToEdit(contributor);
+				handleContributorUpdateModalOpen();
+			}
+		},
+		[handleContributorUpdateModalOpen, projectContributors],
+	);
+
+	const handleContributorUpdateSubmit = useCallback(
+		(payload: ContributorPatchRequestDto) => {
+			if (contributorToEdit) {
+				void dispatch(
+					contributorActions.patch({ id: contributorToEdit.id, payload }),
+				).then(() => {
+					if (projectId) {
+						void dispatch(
+							projectActions.loadAllContributorsByProjectId(projectId),
+						);
+					}
+				});
```

Comment:
we shouldn't await dispatch, dispatch this follow up action inside contributorActions.patch action creator